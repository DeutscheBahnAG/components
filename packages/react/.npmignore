# Documentation
*-doc.js
*-doc.scss
*-doc.css
# Tests
*.snap
*.test.*
#
.release-it.js
tsconfig.build.json
wss_config.yml
