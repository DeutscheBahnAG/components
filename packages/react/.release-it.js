module.exports = {
  git: {
    commitMessage: 'chore: Release @bahn-x/react@${version}',
    requireBranch: 'main',
    tagName: '@bahn-x/react@${version}',
  },
  npm: {
    publishPath: 'dist',
  },
};
