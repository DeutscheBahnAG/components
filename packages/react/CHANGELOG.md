# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

Make sure to format this file using Prettier.

## [Unreleased]

## [1.1.1] - 2022-06-28

### Fixed

- Require latest design fixes of `@db-design/styles`

## [1.1.0] - 2022-06-28

### Added

- Columns support for Container

## [1.0.1] - 2022-05-11

### Fixed

- Missing inline spacer for `<Track>`

## [1.0.0] - 2022-05-10

### Changed

- Renamed `mt`/`mb` to `logoUnitsTop`/`logoUnitsBottom` for `<Logo>` and `<Pulse>`

## [1.0.0-pre.24] - 2022-03-23
