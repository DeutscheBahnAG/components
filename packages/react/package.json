{
  "name": "@db-design/react",
  "version": "1.0.0-pre.19",
  "description": "DB React Components",
  "repository": {
    "directory": "packages/react",
    "url": "git@ssh.git.tech.rz.db.de:bahn-x/webdev/web.git",
    "type": "git"
  },
  "license": "SEE LICENSE IN LICENSE.md",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "dependencies": {
    "@db-design/react-icons": "^1.0.0-pre.1",
    "@db-design/styles": "^1.0.0-pre.17",
    "clsx": "^1.1.1",
    "react": "^17.0.1",
    "react-dom": "^17.0.1",
    "react-modal": "3.12.1"
  },
  "devDependencies": {
    "@testing-library/react": "^12.1.2",
    "@types/react-modal": "^3.10.6",
    "release-it": "^14.2.2",
    "typescript": "^4.1.3"
  },
  "scripts": {
    "build": "rm -rf dist && tsc --project tsconfig.build.json",
    "postbuild": "./postbuild.sh",
    "release": "release-it --disable-metrics"
  },
  "sideEffects": [
    "./dist/button/*",
    "./dist/modal/*",
    "./dist/pulse/*",
    "./dist/triptime/*",
    "./dist/triptimespan/*"
  ]
}
