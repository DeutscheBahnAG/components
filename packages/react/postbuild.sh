#! /bin/bash

cp -p package.json dist &&
for f in `ls *.md`; do cp -p $f dist; done &&
for f in `ls src/*.d.ts`; do cp -p $f dist; done &&
for f in `ls src/*/README.md`; do cp -p $f ${f/src/dist}; done &&
perl -i -pe 's#dist/index#index#' dist/package.json
perl -i -pe 's#dist/module/index#module/index#' dist/package.json
