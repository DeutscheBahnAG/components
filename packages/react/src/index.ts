export * as TransportchipIcons from './transportchip/icons';

export { default as Anchor } from './anchor';
export { default as Button } from './button';
export { default as Checkbox } from './checkbox';
export { default as Container } from './container';
export { default as Copy } from './copy';
export { default as Layout } from './layout';
export { default as Link } from './link';
export { default as Loadingindicator } from './loadingindicator';
export { default as Logo } from './logo';
export { default as Modal } from './modal';
export { default as Notification } from './notification';
export { default as Passwordfield } from './passwordfield';
export { default as Pulse } from './pulse';
export { default as Select } from './select';
export { default as Status } from './status';
export { default as Textfield } from './textfield';
export { default as Theme } from './theme';
export { default as Time } from './time';
export { default as Title } from './title';
export { default as Track } from './track';
export { default as Transportchip } from './transportchip';
export { default as Triptime } from './triptime';
export { default as Triptimespan } from './triptimespan';
export { default as Radiobutton } from './radiobutton';
export { default as Accordion } from './accordion';
export { default as Accordionitem } from './accordionitem';
export { default as List } from './list';
