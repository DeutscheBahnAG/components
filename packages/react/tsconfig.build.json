{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "esModuleInterop": true,
    "jsx": "react",
    "lib": ["dom", "ES2020"],
    "noImplicitAny": false,
    "outDir": "dist",
    "target": "ES2015",
    "sourceMap": false
  },
  "include": ["src"],
  "exclude": ["**/*.test.*", "**/*-doc.*", "**/*.snap"]
}
