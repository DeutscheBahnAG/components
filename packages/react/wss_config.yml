#DISCLAIMER
#All sensible data is not persisted and deleted shortly after usage. So be assured we cope according to any law with your data!

# NOTE: see here how these configs are used https://git.tech.rz.db.de/ass/dependencychecker/whitesource/whitesource-dockered-scanner/-/blob/master/Docker/base/wrapper/scan

##################
#   WHITESOURCE  #
##################
wss_project_name="db-design-react"
wss_project_version="$CI_COMMIT_BRANCH"
  #wss_product_version="latest" DO NOT SET THIS IF USING wss_project_version as this overrides it

  # The following variables must be set as env variables in the Gitlab bahn-x group, they are valid for all subgroups
  #
  # wss_apiKey  -> your organization key, from Whitesource WebGUI:  Integrate > Organization > API key
  # wss_userKey -> CI tech-user key, as received initially via email, also found in 1pass

  # The following variables must be set as env variables in the Gitlab bahn-x subgroups like TMP, Analytics etc
  #
  # wss_product_name  -> set set this to one of our whitesource "products"  https://whitesource.int-app.comp.db.de/Wss/WSS.html#!allProducts
  # wss_product_token -> the products key, from Whitesource WebGUI:  Integrate > Product Tokens > Token

  # Npm specific config: If you have scoped dependencies, please add them according to the conventions below:
  # npmScopedDeps="@yourScope:registry=yourRegistryLink;@yourScope2:registry=yourRegistryLink2"
npmScopedDeps="@bahn-x:registry=https://npm.pkg.github.com"

  # Whitesource will scan your entire project by deafult.
  # Change this, if you want scan only a specific path.
wss_scan_folder=packages/react/

# You feel brave today?
# Great, use your own wss-unified-agent.config file as of here: https://whitesource.atlassian.net/wiki/spaces/WD/pages/489160834/Unified+Agent+Configuration+File+Parameters
# Tell us, where we will find it in your repo structure, starting from the root directory.
# no leading slash!!
#
# !! Please note that in this case all information (including project / product name and version) must be specified in this team-specific Whitesource config file and will no longer be read from wss_config.yml
# (apart from wss_scan_folder and wss_javaParams which is meta-information (i.e. not Whitesource specific) and therefore still needs to be specified).
#wss_specific_team_config=whitesource/wss-unified-agent.config
