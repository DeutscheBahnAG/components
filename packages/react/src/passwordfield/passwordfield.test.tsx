import * as React from 'react';
import { render, screen } from '@testing-library/react';
import Passwordfield from './passwordfield-doc';

const noopFn = () => {};

describe('Passwordfield component', () => {
  it('should render a text input', () => {
    const { container } = render(<Passwordfield value="test123" onChange={noopFn} />);
    expect(screen.getByDisplayValue('test123')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should disable the text input and the button', () => {
    const { container } = render(<Passwordfield disabled value="test456" />);
    const field = screen.getByDisplayValue('test456');
    const button = screen.getByRole('button');
    expect((field as HTMLInputElement).disabled).toBe(true);
    expect((button as HTMLButtonElement).disabled).toBe(true);
    expect(container).toMatchSnapshot();
  });

  it('should disable the button when the value is empty', () => {
    const { container } = render(<Passwordfield value="" />);
    const button = screen.getByRole('button');
    expect(button).toBeDisabled();
    expect(container).toMatchSnapshot();
  });

  it('should enable the button when the value is set', () => {
    const { container } = render(<Passwordfield value="filled" />);
    const button = screen.getByRole('button');
    expect(button).not.toBeDisabled();
    expect(container).toMatchSnapshot();
  });
});
