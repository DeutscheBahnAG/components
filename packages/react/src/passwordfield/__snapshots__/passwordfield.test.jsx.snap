// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Passwordfield component should disable the button when the value is empty 1`] = `
<div>
  <div
    class="db-textfield db-textfield--password db-textfield--size-xl"
  >
    <input
      autocapitalize="none"
      autocomplete="off"
      autocorrect="false"
      spellcheck="false"
      type="password"
      value=""
    />
    <span
      class="db-textfield-suffix"
    >
      <button
        class="db-button db-button--hover-only db-button--round db-button--disabled db-button--size-s"
        disabled=""
        title="Passwort im Klartext anzeigen"
        type="button"
      >
        <svg
          class="db-icon"
          height="24"
          role="img"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 5c3.688 0 6.406 2.188 9.781 6.375A.955.955 0 0122 12a.955.955 0 01-.219.625C18.406 16.844 15.688 19 12 19c-3.469 0-5.938-1.688-9.781-6.375A.955.955 0 012 12c0-.219.063-.438.219-.625C5.594 7.187 8.313 5 12 5zm0 2c-2.531 0-4.688 1.406-7.719 5C7.22 15.5 9.25 16.906 12 17c2.531 0 4.688-1.406 7.719-5C16.78 8.5 14.75 7.094 12 7zm0 2a3 3 0 11-.002 6.002A3 3 0 0112 9zm0 2c-.563 0-1 .469-1 1 0 .563.438 1.031 1 1.031.563 0 1-.469 1-1.031 0-.531-.438-1-1-1z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <span
          aria-hidden="false"
        >
          Passwort im Klartext anzeigen
        </span>
      </button>
      <span
        class="db-inline-spacer"
      >
         
      </span>
    </span>
  </div>
  <span
    class="db-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Passwordfield component should disable the text input and the button 1`] = `
<div>
  <div
    class="db-textfield db-textfield--password db-textfield--size-xl db-textfield--filled db-textfield--disabled"
  >
    <input
      autocapitalize="none"
      autocomplete="off"
      autocorrect="false"
      disabled=""
      spellcheck="false"
      type="password"
      value="test456"
    />
    <span
      class="db-textfield-suffix"
    >
      <button
        class="db-button db-button--hover-only db-button--round db-button--disabled db-button--size-s"
        disabled=""
        title="Passwort im Klartext anzeigen"
        type="button"
      >
        <svg
          class="db-icon"
          height="24"
          role="img"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 5c3.688 0 6.406 2.188 9.781 6.375A.955.955 0 0122 12a.955.955 0 01-.219.625C18.406 16.844 15.688 19 12 19c-3.469 0-5.938-1.688-9.781-6.375A.955.955 0 012 12c0-.219.063-.438.219-.625C5.594 7.187 8.313 5 12 5zm0 2c-2.531 0-4.688 1.406-7.719 5C7.22 15.5 9.25 16.906 12 17c2.531 0 4.688-1.406 7.719-5C16.78 8.5 14.75 7.094 12 7zm0 2a3 3 0 11-.002 6.002A3 3 0 0112 9zm0 2c-.563 0-1 .469-1 1 0 .563.438 1.031 1 1.031.563 0 1-.469 1-1.031 0-.531-.438-1-1-1z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <span
          aria-hidden="false"
        >
          Passwort im Klartext anzeigen
        </span>
      </button>
      <span
        class="db-inline-spacer"
      >
         
      </span>
    </span>
  </div>
  <span
    class="db-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Passwordfield component should enable the button when the value is set 1`] = `
<div>
  <div
    class="db-textfield db-textfield--password db-textfield--size-xl db-textfield--filled"
  >
    <input
      autocapitalize="none"
      autocomplete="off"
      autocorrect="false"
      spellcheck="false"
      type="password"
      value="filled"
    />
    <span
      class="db-textfield-suffix"
    >
      <button
        class="db-button db-button--hover-only db-button--round db-button--size-s"
        title="Passwort im Klartext anzeigen"
        type="button"
      >
        <svg
          class="db-icon"
          height="24"
          role="img"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 5c3.688 0 6.406 2.188 9.781 6.375A.955.955 0 0122 12a.955.955 0 01-.219.625C18.406 16.844 15.688 19 12 19c-3.469 0-5.938-1.688-9.781-6.375A.955.955 0 012 12c0-.219.063-.438.219-.625C5.594 7.187 8.313 5 12 5zm0 2c-2.531 0-4.688 1.406-7.719 5C7.22 15.5 9.25 16.906 12 17c2.531 0 4.688-1.406 7.719-5C16.78 8.5 14.75 7.094 12 7zm0 2a3 3 0 11-.002 6.002A3 3 0 0112 9zm0 2c-.563 0-1 .469-1 1 0 .563.438 1.031 1 1.031.563 0 1-.469 1-1.031 0-.531-.438-1-1-1z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <span
          aria-hidden="false"
        >
          Passwort im Klartext anzeigen
        </span>
      </button>
      <span
        class="db-inline-spacer"
      >
         
      </span>
    </span>
  </div>
  <span
    class="db-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Passwordfield component should render a text input 1`] = `
<div>
  <div
    class="db-textfield db-textfield--password db-textfield--size-xl db-textfield--filled"
  >
    <input
      autocapitalize="none"
      autocomplete="off"
      autocorrect="false"
      spellcheck="false"
      type="password"
      value="test123"
    />
    <span
      class="db-textfield-suffix"
    >
      <button
        class="db-button db-button--hover-only db-button--round db-button--size-s"
        title="Passwort im Klartext anzeigen"
        type="button"
      >
        <svg
          class="db-icon"
          height="24"
          role="img"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 5c3.688 0 6.406 2.188 9.781 6.375A.955.955 0 0122 12a.955.955 0 01-.219.625C18.406 16.844 15.688 19 12 19c-3.469 0-5.938-1.688-9.781-6.375A.955.955 0 012 12c0-.219.063-.438.219-.625C5.594 7.187 8.313 5 12 5zm0 2c-2.531 0-4.688 1.406-7.719 5C7.22 15.5 9.25 16.906 12 17c2.531 0 4.688-1.406 7.719-5C16.78 8.5 14.75 7.094 12 7zm0 2a3 3 0 11-.002 6.002A3 3 0 0112 9zm0 2c-.563 0-1 .469-1 1 0 .563.438 1.031 1 1.031.563 0 1-.469 1-1.031 0-.531-.438-1-1-1z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <span
          aria-hidden="false"
        >
          Passwort im Klartext anzeigen
        </span>
      </button>
      <span
        class="db-inline-spacer"
      >
         
      </span>
    </span>
  </div>
  <span
    class="db-inline-spacer"
  >
     
  </span>
</div>
`;
